src/com/owncloud/android/utils/FileStorageUtils.java

```diff
@@ -154,7 +154,11 @@ public static String getParentPath(String remotePath) {
     public static OCFile fillOCFile(RemoteFile remote) {
         OCFile file = new OCFile(remote.getRemotePath());
         file.setCreationTimestamp(remote.getCreationTimestamp());
-        file.setFileLength(remote.getLength());
+        if (remote.getMimeType().equalsIgnoreCase("DIR")){
+            file.setFileLength(remote.getSize());
```

Comment:
@davivel that's a good question!
For sure, I recall that we don't support oc4, I remember that the entrypoint changed that time (LOL), having said that, I will go with what we support as platform, we are about to stop supporting oc7 https://github.com/owncloud/core/wiki/Maintenance-and-Release-Schedule
@MTRichards is the one